### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by Ashrom0**

### CODE REVIEW

**Review by nwnel-017 — COMMENTED** [review]
I will Review your project

**Review by nwnel-017 — COMMENTED** [review]
Seems pretty unfinished. In the BaseLoggerMixins class, you need to call the log method using the loglevel and message as parameters. Doesn't look like the FileLogger class has been finished, it still needs a constructor, ConfigureFilePath method and Log method. The createLogger method in the log factory needs to be completed too, right now the first line in the method returns null for the logger, but it should only return null if the filepath field is empty. Looks like you need to add unit tests as well

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿[*.cs]
+
+# CS8602: Dereference of a possibly null reference.
+dotnet_diagnostic.CS8602.severity = none
```
Comment (mmwoodfo) [inline .editorconfig:4]: That should not be turned off.

File: Logger/FileLogger.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Logger
+{
+    public class FileLogger: BaseLogger
+    {
+        private string? ClassName { get; set; }
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:10]: This is inherited from `BaseLogger`
```suggestion

```

File: Logger/BaseLogger.cs [hunk 3]
```diff
@@ -3,5 +3,6 @@
     public abstract class BaseLogger
     {
         public abstract void Log(LogLevel logLevel, string message);
+        public string? className { get; set; }
```
Comment (mmwoodfo) [inline Logger/BaseLogger.cs:6]:
```suggestion
        public string? ClassName { get; set; }
```

File: Logger/FileLogger.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        private string? FilePath { get; set; }
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:11]:
```suggestion
        private string? FilePath { get;  }
```
You can make this readonly

File: Logger/FileLogger.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+        public FileLogger(string? className, string? filepath)
+        {
+            ClassName = className;
+            FilePath = filepath;
+        }
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:17]:
```suggestion
        public FileLogger(string filepath)
        {
            FilePath = filepath;
        }
```

File: Logger/FileLogger.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+        public override void Log(LogLevel logLevel, string message)
+        {
+            using StreamWriter file = new(FilePath, append: true);
+            string currentTime = DateTime.Now.ToString();
+            string log = $"{currentTime} {ClassName} {logLevel}: {message}";
+            file.WriteLine(log);
```
Comment (mmwoodfo) [inline Logger/FileLogger.cs:24]:
```suggestion
            string log = $"{DateTime.Now} {ClassName} {logLevel}: {message}";
            file.WriteLine(log);
```

Could also do this to get date time to a string - but either way works just fine!

File: Logger/LogFactory.cs [hunk 7]
```diff
@@ -1,11 +1,22 @@
-﻿namespace Logger
+﻿using System;
+using System.IO;
+
+namespace Logger
 {
     public class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private string? filepath { get; set; }
```
Comment (mmwoodfo) [inline Logger/LogFactory.cs:8]:
```suggestion
        private string? Filepath { get; set; }
```

File: Logger/LogFactory.cs [hunk 8]
```diff
@@ -1,11 +1,22 @@
...
         {
-            
+            if (filepath != null)
+                return new FileLogger(className,filepath);
             return null;
         }
+
+        public void ConfigureFileLogger(string filePath)
+        {
+            filepath = filePath;
```
Comment (mmwoodfo) [inline Logger/LogFactory.cs:19]:
```suggestion
            Filepath = filePath;
```

File: Logger.Tests/BaseLoggerMixinsTests.cs [hunk 9]
```diff
@@ -14,7 +14,7 @@ public void Error_WithNullLogger_ThrowsException()
             // Arrange
 
             // Act
-            //BaseLoggerMixins.Error(null, "");
```
Comment (mmwoodfo) [inline Logger.Tests/BaseLoggerMixinsTests.cs:17]: You are missing alot of tests.

**Pull request closed without merging**